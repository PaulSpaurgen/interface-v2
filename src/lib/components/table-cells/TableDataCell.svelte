<script lang="ts">
	import { tableCellClasses } from '$lib/atoms/componentClasses';
	export let styleClass = '';
</script>

<td class="{styleClass} {tableCellClasses.rowNormal}">
	<slot />
</td>
