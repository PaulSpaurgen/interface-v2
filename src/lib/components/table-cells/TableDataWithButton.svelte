<script>
	import { tableCellClasses } from '$lib/atoms/componentClasses';
</script>

<td class={tableCellClasses.row}>
	<div class={tableCellClasses.rowWithIcon}>
		<slot />
	</div>
</td>
