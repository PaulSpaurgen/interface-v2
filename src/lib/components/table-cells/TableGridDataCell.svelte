<script lang="ts">
	import { tableCellClasses } from '$lib/atoms/componentClasses';
	export let styleClass = '';
	export let width = '';
</script>

<div class={`${styleClass} ${tableCellClasses.rowNormal}`} style={`width:${width}`}>
	<slot />
</div>
