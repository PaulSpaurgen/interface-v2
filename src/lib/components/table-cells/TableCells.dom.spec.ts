import { cleanup, render } from '@testing-library/svelte';
import TableDataCell from './TableDataCell.svelte';
import html from 'svelte-htm';
import TableGridDataCell from './TableGridDataCell.svelte';
import TableDataWithButton from './TableDataWithButton.svelte';

describe('Table Cells', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	test('TableDataCell component renders ', () => {
		const { getByTestId, container } = render(TableDataCell, {
			styleClass: 'table-data-test-class'
		});
		expect(getByTestId('table-data-cell')).toBeTruthy();
		expect(getByTestId('table-data-cell').classList.contains('table-data-test-class')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	test('TableDataCell component renders with slot ', () => {
		const { getByTestId, getByLabelText } = render(
			html`<${TableDataCell}><div aria-label="test-component-chidren">Test Component Chidren</div></${TableDataCell}>`
		);
		expect(getByTestId('table-data-cell')).toBeTruthy();
		expect(
			getByTestId('table-data-cell').contains(getByLabelText('test-component-chidren'))
		).toBeTruthy();

		expect(
			getByLabelText('test-component-chidren').innerHTML === 'Test Component Chidren'
		).toBeTruthy();
	});

	test('TableGridDataCell component renders ', () => {
		const { getByTestId, container } = render(TableGridDataCell, {
			styleClass: 'table-grid-data-test-class'
		});
		expect(getByTestId('table-grid-data-cell')).toBeTruthy();
		expect(
			getByTestId('table-grid-data-cell').classList.contains('table-grid-data-test-class')
		).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	test('TableGridDataCell component renders with slot ', () => {
		const { getByTestId, getByLabelText } = render(
			html`<${TableGridDataCell}><div aria-label="test-component-chidren">Test Component Chidren</div></${TableGridDataCell}>`
		);
		expect(getByTestId('table-grid-data-cell')).toBeTruthy();
		expect(
			getByTestId('table-grid-data-cell').contains(getByLabelText('test-component-chidren'))
		).toBeTruthy();

		expect(
			getByLabelText('test-component-chidren').innerHTML === 'Test Component Chidren'
		).toBeTruthy();
	});

	test('TableDataWithButton component renders ', () => {
		const { getByTestId, container } = render(TableDataWithButton, {
			styleClass: 'table-data-with-button-test-class'
		});
		expect(getByTestId('table-data-with-button')).toBeTruthy();
		expect(
			getByTestId('table-data-with-button').classList.contains('table-data-with-button-test-class')
		).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	test('TableDataWithButton component renders with slots', () => {
		const { getByTestId, findByText } = render(
			html`<${TableDataWithButton}>
            <button slot='line1'>Test Component First Chidren</button>
            <button slot='line2'>Test Component Second Chidren</button>
            </${TableDataWithButton}>`
		);
		expect(getByTestId('table-data-with-button')).toBeTruthy();
		expect(findByText('Test Component First Chidren')).toBeTruthy();
		expect(findByText('Test Component Second Chidren')).toBeTruthy();
	});
});
