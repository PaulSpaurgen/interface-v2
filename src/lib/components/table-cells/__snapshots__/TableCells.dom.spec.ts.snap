// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Cells > TableDataCell component renders  1`] = `
<body>
  <div>
    <td
      class="table-data-test-class font-normal text-sm py-3 whitespace-nowrap"
      data-testid="table-data-cell"
    />
    <!--&lt;TableDataCell&gt;-->
  </div>
</body>
`;

exports[`Table Cells > TableDataWithButton component renders  1`] = `
<body>
  <div>
    <td
      class="pb-8 pt-6 table-data-with-button-test-class font-semibold text-base py-6 whitespace-nowrap"
      data-testid="table-data-with-button"
    >
      <div
        class="flex flex-col items-center justify-start"
      >
        <div
          class="m-auto flex h-[35px] items-center"
        />
         
        <div
          class="flex h-[18px] items-center pt-[10px]"
        />
      </div>
    </td>
    <!--&lt;TableDataWithButton&gt;-->
  </div>
</body>
`;

exports[`Table Cells > TableGridDataCell component renders  1`] = `
<body>
  <div>
    <div
      class="table-grid-data-test-class font-normal text-sm py-3 whitespace-nowrap"
      data-testid="table-grid-data-cell"
    />
    <!--&lt;TableGridDataCell&gt;-->
  </div>
</body>
`;
